AdobeMediaGallery/Test/Unit/Model/Asset/Command/DeleteByIdTest.php

```diff
@@ -0,0 +1,177 @@
+use Magento\AdobeMediaGalleryApi\Api\Data\AssetInterfaceFactory;
+use Magento\AdobeMediaGalleryApi\Model\Asset\Command\GetByIdInterface;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\DB\Adapter\AdapterInterface;
+use Magento\Framework\Exception\CouldNotDeleteException;
+use PHPUnit\Framework\TestCase;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Test the DeleteById command model
+ */
+class DeleteByIdTest extends TestCase
+{
+    /**
+     * @var DeleteById
+     */
+    private $deleteMediaAssetById;
+
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+
+    /**
+     * @var GetByIdInterface
+     */
+    private $getMediaAssetById;
+
+    /**
+     * @var AssetInterfaceFactory
+     */
+    private $mediaAssetFactory;
+
+    /**
+     * @var AssetInterface
+     */
+    private $mediaAsset;
+
+    /**
+     * @var AdapterInterface
+     */
+    private $adapter;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * Initialize basic test class mocks
+     */
+    protected function setUp()
+    {
+        $this->resourceConnection = $this->createMock(ResourceConnection::class);
+        $this->getMediaAssetById = $this->createMock(GetByIdInterface::class);
+        $this->mediaAssetFactory = $this->createMock(AssetInterfaceFactory::class);
+        $this->logger = $this->createMock(LoggerInterface::class);
+
+        $this->deleteMediaAssetById = (new ObjectManager($this))->getObject(
+            DeleteById::class,
+            [
+                'resourceConnection' => $this->resourceConnection,
+                'getMediaAssetById' => $this->getMediaAssetById,
+                'mediaAssetFactory' =>  $this->mediaAssetFactory,
+                'logger' =>  $this->logger,
+            ]
+        );
+
+        $this->mediaAsset = $this->createMock(AssetInterface::class);
+        $this->adapter = $this->createMock(AdapterInterface::class);
+    }
+
+    /**
+     * Test delete media asset by id action
+     *
+     * @param int $mediaAssetId
+     * @dataProvider deleteMediaAssetTestScopes
+     */
+    public function testSuccessfulDeleteByIdExecution(int $mediaAssetId)
+    {
+        $this->getMediaAssetById->expects($this->at(0))
+            ->method('execute')
+            ->with($mediaAssetId)
+            ->willReturn($this->mediaAsset);
+
+        $this->resourceConnection->expects($this->at(0))
+            ->method('getConnection')
+            ->willReturn($this->adapter);
+
+        $tableName = 'media_gallery_asset';
+        $this->resourceConnection->expects($this->at(1))
+            ->method('getTableName')
+            ->with($tableName)
+            ->willReturn($tableName);
+
+        $this->mediaAsset->expects($this->once())
+            ->method('getId')
+            ->willReturn($mediaAssetId);
+
+        $this->adapter->expects($this->at(0))
```

Comment:
There is only one call to `adapter::delete` method expected, no need to verify the sequence
```suggestion
        $this->adapter->expects($this->once())
```